### PR TITLE
Waffling60 Rev E ANSI Hotswap

### PR DESCRIPTION
**Pull request by 4pplet**
## Description

Adding support for the latest revision (E) of Waffling60 Hotswap. Same layout support but adds underglow and rotary encoder support.

Source for PCB is available and manufacturing files are available as pre-release (https://github.com/4pplet/waffling60)

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/4pplet/waffling60/rev_e_ansi/matrix_diagram.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴─────┤  
+│20    │21 │22 │23 │24 │25 │26 │27 │28 │29 │2A │2B │2D      │   
+├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────┬───┤ 
+│30      │32 │33 │34 │35 │36 │37 │38 │39 │3A │3B │3C    │3D │   
+├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴───┴───┼───┴┬────┬┴───┤
+│40  │41  │42  │46                           │4B  │4C  │4D  │ Tsangan/WKL
+└────┴────┴────┴─────────────────────────────┴────┴────┴────┘
+┌─────┬───┬─────┬──────────┬──────┬─────────┬─────┬───┬─────┐
+│40   │41 │42   │44        │46    │48       │4B   │4C │4D   │ 3U split space
+└─────┴───┴─────┴──────────┴──────┴─────────┴─────┴───┴─────┘
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e_ansi/matrix_diagram.md:20]:
```suggestion
┌─────┬───┬─────┬───────────┬───┬───────────┬─────┬───┬─────┐
│40   │41 │42   │44         │46 │48         │4B   │4C │4D   │ 7u split spacebar
└─────┴───┴─────┴───────────┴───┴───────────┴─────┴───┴─────┘
```

File: keyboards/4pplet/waffling60/rev_e_ansi/matrix_diagram.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┐
+│00 │01 │02 │03 │04 │05 │06 │07 │08 │09 │0A │0B │0C │0D │1D │
+├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┤      
+│10   │11 │12 │13 │14 │15 │16 │17 │18 │19 │1A │1B │1C │2C   │      
+├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴─────┤  
+│20    │21 │22 │23 │24 │25 │26 │27 │28 │29 │2A │2B │2D      │   
+├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────┬───┤ 
+│30      │32 │33 │34 │35 │36 │37 │38 │39 │3A │3B │3C    │3D │   
+├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴───┴───┼───┴┬────┬┴───┤
+│40  │41  │42  │46                           │4B  │4C  │4D  │ Tsangan/WKL
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e_ansi/matrix_diagram.md:16]:
```suggestion
│40  │41  │42  │46                           │4B  │4C  │4D  │ Tsangan/WKL/HHKB
```

File: keyboards/4pplet/waffling60/rev_e_ansi/info.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                {"label": "Fn", "matrix": [3, 13], "x": 14, "y": 3},
+                {"label": "Ctrl", "matrix": [4, 0], "x": 0, "y": 4, "w": 1.5},
+                {"label": "Win", "matrix": [4, 1], "x": 1.5, "y": 4},
+                {"label": "Alt", "matrix": [4, 2], "x": 2.5, "y": 4, "w": 1.5},
+                {"label": "Space", "matrix": [4, 4], "x":4, "y":4, "w":3},
+                {"label": "Space", "matrix": [4, 6], "x":7, "y":4},
+                {"label": "Space", "matrix": [4, 8], "x":8, "y":4, "w":3},
+                {"label": "Alt", "matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
+                {"label": "Win", "matrix": [4, 12], "x": 12.5, "y": 4},
+                {"label": "Ctrl", "matrix": [4, 13], "x": 13.5, "y": 4, "w": 1.5}
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e_ansi/info.json:117]:
```suggestion
    },
    "community_layouts": [
        "60_tsangan_hhkb",
        "60_hhkb"
    ],
    "layouts": {
        "LAYOUT_all": {
            "layout": [
                {"label": "Esc", "matrix": [0, 0], "x": 0, "y": 0},
                {"label": "!", "matrix": [0, 1], "x": 1, "y": 0},
                {"label": "@", "matrix": [0, 2], "x": 2, "y": 0},
                {"label": "#", "matrix": [0, 3], "x": 3, "y": 0},
                {"label": "$", "matrix": [0, 4], "x": 4, "y": 0},
                {"label": "%", "matrix": [0, 5], "x": 5, "y": 0},
                {"label": "^", "matrix": [0, 6], "x": 6, "y": 0},
                {"label": "&", "matrix": [0, 7], "x": 7, "y": 0},
                {"label": "*", "matrix": [0, 8], "x": 8, "y": 0},
                {"label": "(", "matrix": [0, 9], "x": 9, "y": 0},
                {"label": ")", "matrix": [0, 10], "x": 10, "y": 0},
                {"label": "_", "matrix": [0, 11], "x": 11, "y": 0},
                {"label": "+", "matrix": [0, 12], "x": 12, "y": 0},
                {"label": "|", "matrix": [0, 13], "x": 13, "y": 0},
                {"label": "~", "matrix": [1, 13], "x": 14, "y": 0},
                {"label": "Tab", "matrix": [1, 0], "x": 0, "y": 1, "w": 1.5},
                {"label": "Q", "matrix": [1, 1], "x": 1.5, "y": 1},
                {"label": "W", "matrix": [1, 2], "x": 2.5, "y": 1},
                {"label": "E", "matrix": [1, 3], "x": 3.5, "y": 1},
                {"label": "R", "matrix": [1, 4], "x": 4.5, "y": 1},
                {"label": "T", "matrix": [1, 5], "x": 5.5, "y": 1},
                {"label": "Y", "matrix": [1, 6], "x": 6.5, "y": 1},
                {"label": "U", "matrix": [1, 7], "x": 7.5, "y": 1},
                {"label": "I", "matrix": [1, 8], "x": 8.5, "y": 1},
                {"label": "O", "matrix": [1, 9], "x": 9.5, "y": 1},
                {"label": "P", "matrix": [1, 10], "x": 10.5, "y": 1},
                {"label": "{", "matrix": [1, 11], "x": 11.5, "y": 1},
                {"label": "}", "matrix": [1, 12], "x": 12.5, "y": 1},
                {"label": "|", "matrix": [2, 12], "x": 13.5, "y": 1, "w": 1.5},
                {"label": "Caps Lock", "matrix": [2, 0], "x": 0, "y": 2, "w": 1.75},
                {"label": "A", "matrix": [2, 1], "x": 1.75, "y": 2},
                {"label": "S", "matrix": [2, 2], "x": 2.75, "y": 2},
                {"label": "D", "matrix": [2, 3], "x": 3.75, "y": 2},
                {"label": "F", "matrix": [2, 4], "x": 4.75, "y": 2},
                {"label": "G", "matrix": [2, 5], "x": 5.75, "y": 2},
                {"label": "H", "matrix": [2, 6], "x": 6.75, "y": 2},
                {"label": "J", "matrix": [2, 7], "x": 7.75, "y": 2},
                {"label": "K", "matrix": [2, 8], "x": 8.75, "y": 2},
                {"label": "L", "matrix": [2, 9], "x": 9.75, "y": 2},
                {"label": ":", "matrix": [2, 10], "x": 10.75, "y": 2},
                {"label": "\"", "matrix": [2, 11], "x": 11.75, "y": 2},
                {"label": "Enter", "matrix": [2, 13], "x": 12.75, "y": 2, "w": 2.25},
                {"label": "Shift", "matrix": [3, 0], "x": 0, "y": 3, "w": 2.25},
                {"label": "Z", "matrix": [3, 2], "x": 2.25, "y": 3},
                {"label": "X", "matrix": [3, 3], "x": 3.25, "y": 3},
                {"label": "C", "matrix": [3, 4], "x": 4.25, "y": 3},
                {"label": "V", "matrix": [3, 5], "x": 5.25, "y": 3},
                {"label": "B", "matrix": [3, 6], "x": 6.25, "y": 3},
                {"label": "N", "matrix": [3, 7], "x": 7.25, "y": 3},
                {"label": "M", "matrix": [3, 8], "x": 8.25, "y": 3},
                {"label": "<", "matrix": [3, 9], "x": 9.25, "y": 3},
                {"label": ">", "matrix": [3, 10], "x": 10.25, "y": 3},
                {"label": "?", "matrix": [3, 11], "x": 11.25, "y": 3},
                {"label": "Shift", "matrix": [3, 12], "x": 12.25, "y": 3, "w": 1.75},
                {"label": "Fn", "matrix": [3, 13], "x": 14, "y": 3},
                {"label": "Ctrl", "matrix": [4, 0], "x": 0, "y": 4, "w": 1.5},
                {"label": "Win", "matrix": [4, 1], "x": 1.5, "y": 4},
                {"label": "Alt", "matrix": [4, 2], "x": 2.5, "y": 4, "w": 1.5},
                {"label": "Space", "matrix": [4, 4], "x":4, "y":4, "w":3},
                {"label": "Space", "matrix": [4, 6], "x":7, "y":4},
                {"label": "Space", "matrix": [4, 8], "x":8, "y":4, "w":3},
                {"label": "Alt", "matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
                {"label": "Win", "matrix": [4, 12], "x": 12.5, "y": 4},
                {"label": "Ctrl", "matrix": [4, 13], "x": 13.5, "y": 4, "w": 1.5}
            ]
        },
        "LAYOUT_60_tsangan_hhkb": {
            "layout": [
                {"label": "Esc", "matrix": [0, 0], "x": 0, "y": 0},
                {"label": "!", "matrix": [0, 1], "x": 1, "y": 0},
                {"label": "@", "matrix": [0, 2], "x": 2, "y": 0},
                {"label": "#", "matrix": [0, 3], "x": 3, "y": 0},
                {"label": "$", "matrix": [0, 4], "x": 4, "y": 0},
                {"label": "%", "matrix": [0, 5], "x": 5, "y": 0},
                {"label": "^", "matrix": [0, 6], "x": 6, "y": 0},
                {"label": "&", "matrix": [0, 7], "x": 7, "y": 0},
                {"label": "*", "matrix": [0, 8], "x": 8, "y": 0},
                {"label": "(", "matrix": [0, 9], "x": 9, "y": 0},
                {"label": ")", "matrix": [0, 10], "x": 10, "y": 0},
                {"label": "_", "matrix": [0, 11], "x": 11, "y": 0},
                {"label": "+", "matrix": [0, 12], "x": 12, "y": 0},
                {"label": "|", "matrix": [0, 13], "x": 13, "y": 0},
                {"label": "~", "matrix": [1, 13], "x": 14, "y": 0},
                {"label": "Tab", "matrix": [1, 0], "x": 0, "y": 1, "w": 1.5},
                {"label": "Q", "matrix": [1, 1], "x": 1.5, "y": 1},
                {"label": "W", "matrix": [1, 2], "x": 2.5, "y": 1},
                {"label": "E", "matrix": [1, 3], "x": 3.5, "y": 1},
                {"label": "R", "matrix": [1, 4], "x": 4.5, "y": 1},
                {"label": "T", "matrix": [1, 5], "x": 5.5, "y": 1},
                {"label": "Y", "matrix": [1, 6], "x": 6.5, "y": 1},
                {"label": "U", "matrix": [1, 7], "x": 7.5, "y": 1},
                {"label": "I", "matrix": [1, 8], "x": 8.5, "y": 1},
                {"label": "O", "matrix": [1, 9], "x": 9.5, "y": 1},
                {"label": "P", "matrix": [1, 10], "x": 10.5, "y": 1},
                {"label": "{", "matrix": [1, 11], "x": 11.5, "y": 1},
                {"label": "}", "matrix": [1, 12], "x": 12.5, "y": 1},
                {"label": "|", "matrix": [2, 12], "x": 13.5, "y": 1, "w": 1.5},
                {"label": "Caps Lock", "matrix": [2, 0], "x": 0, "y": 2, "w": 1.75},
                {"label": "A", "matrix": [2, 1], "x": 1.75, "y": 2},
                {"label": "S", "matrix": [2, 2], "x": 2.75, "y": 2},
                {"label": "D", "matrix": [2, 3], "x": 3.75, "y": 2},
                {"label": "F", "matrix": [2, 4], "x": 4.75, "y": 2},
                {"label": "G", "matrix": [2, 5], "x": 5.75, "y": 2},
                {"label": "H", "matrix": [2, 6], "x": 6.75, "y": 2},
                {"label": "J", "matrix": [2, 7], "x": 7.75, "y": 2},
                {"label": "K", "matrix": [2, 8], "x": 8.75, "y": 2},
                {"label": "L", "matrix": [2, 9], "x": 9.75, "y": 2},
                {"label": ":", "matrix": [2, 10], "x": 10.75, "y": 2},
                {"label": "\"", "matrix": [2, 11], "x": 11.75, "y": 2},
                {"label": "Enter", "matrix": [2, 13], "x": 12.75, "y": 2, "w": 2.25},
                {"label": "Shift", "matrix": [3, 0], "x": 0, "y": 3, "w": 2.25},
                {"label": "Z", "matrix": [3, 2], "x": 2.25, "y": 3},
                {"label": "X", "matrix": [3, 3], "x": 3.25, "y": 3},
                {"label": "C", "matrix": [3, 4], "x": 4.25, "y": 3},
                {"label": "V", "matrix": [3, 5], "x": 5.25, "y": 3},
                {"label": "B", "matrix": [3, 6], "x": 6.25, "y": 3},
                {"label": "N", "matrix": [3, 7], "x": 7.25, "y": 3},
                {"label": "M", "matrix": [3, 8], "x": 8.25, "y": 3},
                {"label": "<", "matrix": [3, 9], "x": 9.25, "y": 3},
                {"label": ">", "matrix": [3, 10], "x": 10.25, "y": 3},
                {"label": "?", "matrix": [3, 11], "x": 11.25, "y": 3},
                {"label": "Shift", "matrix": [3, 12], "x": 12.25, "y": 3, "w": 1.75},
                {"label": "Fn", "matrix": [3, 13], "x": 14, "y": 3},
                {"label": "Ctrl", "matrix": [4, 0], "x": 0, "y": 4, "w": 1.5},
                {"label": "Win", "matrix": [4, 1], "x": 1.5, "y": 4},
                {"label": "Alt", "matrix": [4, 2], "x": 2.5, "y": 4, "w": 1.5},
                {"label": "Space", "matrix": [4, 6], "x":4, "y":4, "w": 7},
                {"label": "Alt", "matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
                {"label": "Win", "matrix": [4, 12], "x": 12.5, "y": 4},
                {"label": "Ctrl", "matrix": [4, 13], "x": 13.5, "y": 4, "w": 1.5}
            ]
        },
        "LAYOUT_60_hhkb": {
            "layout": [
                {"label": "Esc", "matrix": [0, 0], "x": 0, "y": 0},
                {"label": "!", "matrix": [0, 1], "x": 1, "y": 0},
                {"label": "@", "matrix": [0, 2], "x": 2, "y": 0},
                {"label": "#", "matrix": [0, 3], "x": 3, "y": 0},
                {"label": "$", "matrix": [0, 4], "x": 4, "y": 0},
                {"label": "%", "matrix": [0, 5], "x": 5, "y": 0},
                {"label": "^", "matrix": [0, 6], "x": 6, "y": 0},
                {"label": "&", "matrix": [0, 7], "x": 7, "y": 0},
                {"label": "*", "matrix": [0, 8], "x": 8, "y": 0},
                {"label": "(", "matrix": [0, 9], "x": 9, "y": 0},
                {"label": ")", "matrix": [0, 10], "x": 10, "y": 0},
                {"label": "_", "matrix": [0, 11], "x": 11, "y": 0},
                {"label": "+", "matrix": [0, 12], "x": 12, "y": 0},
                {"label": "|", "matrix": [0, 13], "x": 13, "y": 0},
                {"label": "~", "matrix": [1, 13], "x": 14, "y": 0},
                {"label": "Tab", "matrix": [1, 0], "x": 0, "y": 1, "w": 1.5},
                {"label": "Q", "matrix": [1, 1], "x": 1.5, "y": 1},
                {"label": "W", "matrix": [1, 2], "x": 2.5, "y": 1},
                {"label": "E", "matrix": [1, 3], "x": 3.5, "y": 1},
                {"label": "R", "matrix": [1, 4], "x": 4.5, "y": 1},
                {"label": "T", "matrix": [1, 5], "x": 5.5, "y": 1},
                {"label": "Y", "matrix": [1, 6], "x": 6.5, "y": 1},
                {"label": "U", "matrix": [1, 7], "x": 7.5, "y": 1},
                {"label": "I", "matrix": [1, 8], "x": 8.5, "y": 1},
                {"label": "O", "matrix": [1, 9], "x": 9.5, "y": 1},
                {"label": "P", "matrix": [1, 10], "x": 10.5, "y": 1},
                {"label": "{", "matrix": [1, 11], "x": 11.5, "y": 1},
                {"label": "}", "matrix": [1, 12], "x": 12.5, "y": 1},
                {"label": "Bsp", "matrix": [2, 12], "x": 13.5, "y": 1, "w": 1.5},
                {"label": "Ctrl", "matrix": [2, 0], "x": 0, "y": 2, "w": 1.75},
                {"label": "A", "matrix": [2, 1], "x": 1.75, "y": 2},
                {"label": "S", "matrix": [2, 2], "x": 2.75, "y": 2},
                {"label": "D", "matrix": [2, 3], "x": 3.75, "y": 2},
                {"label": "F", "matrix": [2, 4], "x": 4.75, "y": 2},
                {"label": "G", "matrix": [2, 5], "x": 5.75, "y": 2},
                {"label": "H", "matrix": [2, 6], "x": 6.75, "y": 2},
                {"label": "J", "matrix": [2, 7], "x": 7.75, "y": 2},
                {"label": "K", "matrix": [2, 8], "x": 8.75, "y": 2},
                {"label": "L", "matrix": [2, 9], "x": 9.75, "y": 2},
                {"label": ":", "matrix": [2, 10], "x": 10.75, "y": 2},
                {"label": "\"", "matrix": [2, 11], "x": 11.75, "y": 2},
                {"label": "Enter", "matrix": [2, 13], "x": 12.75, "y": 2, "w": 2.25},
                {"label": "Shift", "matrix": [3, 0], "x": 0, "y": 3, "w": 2.25},
                {"label": "Z", "matrix": [3, 2], "x": 2.25, "y": 3},
                {"label": "X", "matrix": [3, 3], "x": 3.25, "y": 3},
                {"label": "C", "matrix": [3, 4], "x": 4.25, "y": 3},
                {"label": "V", "matrix": [3, 5], "x": 5.25, "y": 3},
                {"label": "B", "matrix": [3, 6], "x": 6.25, "y": 3},
                {"label": "N", "matrix": [3, 7], "x": 7.25, "y": 3},
                {"label": "M", "matrix": [3, 8], "x": 8.25, "y": 3},
                {"label": "<", "matrix": [3, 9], "x": 9.25, "y": 3},
                {"label": ">", "matrix": [3, 10], "x": 10.25, "y": 3},
                {"label": "?", "matrix": [3, 11], "x": 11.25, "y": 3},
                {"label": "Shift", "matrix": [3, 12], "x": 12.25, "y": 3, "w": 1.75},
                {"label": "Fn", "matrix": [3, 13], "x": 14, "y": 3},
                {"label": "Alt", "matrix": [4, 1], "x": 1.5, "y": 4},
                {"label": "Win", "matrix": [4, 2], "x": 2.5, "y": 4, "w": 1.5},
                {"label": "Space", "matrix": [4, 6], "x":4, "y":4, "w": 7},
                {"label": "Win", "matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
                {"label": "Alt", "matrix": [4, 12], "x": 12.5, "y": 4},
```

File: keyboards/4pplet/waffling60/rev_e_ansi/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+// main layer
+[0] = LAYOUT(
+  KC_ESC,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,       KC_6,      KC_7,      KC_8,      KC_9,       KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_BSPC,
+  KC_TAB,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,       KC_Y,      KC_U,      KC_I,      KC_O,       KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,       KC_H,      KC_J,      KC_K,      KC_L,       KC_SCLN, KC_QUOT, KC_ENT,
+  KC_LSFT,            KC_Z,     KC_X,     KC_C,     KC_V,       KC_B,      KC_N,      KC_M,      KC_COMM,    KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+  KC_LCTL,  KC_LGUI,  KC_LALT,            KC_SPC,               KC_SPC,               KC_SPC,                KC_RALT, KC_RGUI, KC_RCTL),
+// basic function layer
+[1] = LAYOUT(
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e_ansi/keymaps/default/keymap.c:28]:
```suggestion
[0] = LAYOUT_all(
  KC_ESC,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,       KC_6,      KC_7,      KC_8,      KC_9,       KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_BSPC,
  KC_TAB,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,       KC_Y,      KC_U,      KC_I,      KC_O,       KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,       KC_H,      KC_J,      KC_K,      KC_L,       KC_SCLN, KC_QUOT, KC_ENT,
  KC_LSFT,            KC_Z,     KC_X,     KC_C,     KC_V,       KC_B,      KC_N,      KC_M,      KC_COMM,    KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
  KC_LCTL,  KC_LGUI,  KC_LALT,            KC_SPC,               KC_SPC,               KC_SPC,                KC_RALT, KC_RGUI, KC_RCTL),
// basic function layer
[1] = LAYOUT_all(
```

File: keyboards/4pplet/waffling60/rev_e_ansi/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+// main layer
+[0] = LAYOUT(
+  KC_ESC,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,       KC_6,      KC_7,      KC_8,      KC_9,       KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_BSPC,
+  KC_TAB,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,       KC_Y,      KC_U,      KC_I,      KC_O,       KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,       KC_H,      KC_J,      KC_K,      KC_L,       KC_SCLN, KC_QUOT, KC_ENT,
+  KC_LSFT,            KC_Z,     KC_X,     KC_C,     KC_V,       KC_B,      KC_N,      KC_M,      KC_COMM,    KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+  KC_LCTL,  KC_LGUI,  KC_LALT,            KC_SPC,               KC_SPC,               KC_SPC,                KC_RALT, KC_RGUI, KC_RCTL),
+// basic function layer
+[1] = LAYOUT(
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e_ansi/keymaps/via/keymap.c:28]:
```suggestion
[0] = LAYOUT_all(
  KC_ESC,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,       KC_6,      KC_7,      KC_8,      KC_9,       KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_BSPC,
  KC_TAB,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,       KC_Y,      KC_U,      KC_I,      KC_O,       KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,       KC_H,      KC_J,      KC_K,      KC_L,       KC_SCLN, KC_QUOT, KC_ENT,
  KC_LSFT,            KC_Z,     KC_X,     KC_C,     KC_V,       KC_B,      KC_N,      KC_M,      KC_COMM,    KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
  KC_LCTL,  KC_LGUI,  KC_LALT,            KC_SPC,               KC_SPC,               KC_SPC,                KC_RALT, KC_RGUI, KC_RCTL),
// basic function layer
[1] = LAYOUT_all(
```

File: keyboards/4pplet/waffling60/rev_e_ansi/readme.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# waffling60
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e_ansi/readme.md:1]:
```suggestion
# waffling60 Rev. E ANSI
```

File: keyboards/4pplet/waffling60/rev_e_ansi/readme.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+More info: https://github.com/4pplet/waffling60
+
+* Keyboard Maintainer: [4pplet](https://github.com/4pplet)
+* Hardware Supported: [waffling60](https://github.com/4pplet/waffling60)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make 4pplet/waffling60/rev_e_ansi:default
+
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e_ansi/readme.md:13]:
```suggestion

Flashing example for this keyboard:

    make 4pplet/waffling60/rev_e_ansi:default:flash

```

File: keyboards/4pplet/waffling60/rev_e_ansi/readme.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e_ansi/readme.md:15]:
```suggestion

## Bootloader

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by dunk2k — APPROVED** [review]

**Comment by dunk2k** [comment]
@4pplet 
Quick note to say thanks for supplying a `matrix_diagram.md` - helps with Community Layout support and for users that don't/can't use 3rd Party GUI based interfaces for QMK.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**